moveit_plugins/moveit_ros_control_interface/src/controller_manager_plugin.cpp

```diff
@@ -62,18 +67,32 @@ namespace moveit_ros_control_interface
  * @param[in] force force timeout
  * @return True if timeout duration was passed
  */
-bool checkTimeout(ros::Time& t, double timeout, bool force = false)
+bool checkTimeout(const rclcpp::Clock::SharedPtr& clock, rclcpp::Time& t, double timeout, bool force = false)
 {
-  ros::Time now = ros::Time::now();
-  if (force || (now - t) >= ros::Duration(timeout))
+  rclcpp::Time now = clock->now();
+  if (force || (now - t) >= rclcpp::Duration::from_seconds(timeout))
   {
     t = now;
     return true;
   }
   return false;
 }
 
-MOVEIT_CLASS_FORWARD(MoveItControllerManager)  // Defines MoveItControllerManagerPtr, ConstPtr, WeakPtr... etc
+/**
+ * \brief Get joint name from resource name reported by ros2_control, since claimed_interfaces return by ros2_control
+ * will have the interface name as suffix joint_name/INTERFACE_TYPE
+ * @param[in] claimed_interface claimed interface as joint_name/INTERFACE_TYPE
+ * @return joint_name part of the /p claimed_interface
+ */
+std::string getJointName(const std::string& claimed_interface)
+{
+  const auto index = claimed_interface.find('/');
+  if (index == std::string::npos)
+    return claimed_interface;
```

Comment:
should this be handled in any way?

Reply:
Just in case, do you want me to throw an exception if this happens .?

Reply:
I think that would make a lot of sense. Especially if we don't have a better way to validate the resource at this point.